lib/tree.rb

```diff
@@ -104,22 +124,48 @@ def self.find_tree(repository, file_path, revstring=Constants::HEAD)
 
     private
 
-    def jtree_entries(options={})
+    def contents_array
+      @contents_ary ||= jtree_entries
+    end
+
+    def init_walk(recurse=false)
       treewalk = TreeWalk.new(@jrepo)
       treewalk.add_tree(@jtree)
-      treewalk.set_recursive(true) if options[:recursive]
+      treewalk.set_recursive(true) if recurse
+      treewalk
+    end
+
+    def tree_entry(treewalk)
```

Comment:
This generates a `Hash` reprsentation of the entry (not yet a `Blob` or `Tree` object to be yielded in the `#find` methods. So the user can do: `tree.find {|entry| entry[:mode] == foo && entry[:name] == 'bar' }`.

Reply:
Cool!